### PR TITLE
Avoid printing blank lines between "Another exception was thrown:" messages

### PR DESCRIPTION
**Pull request by Hixie**
This also fixes a bug in the test that it modifies. It turns out the test wasn't testing what it thought it was testing, because the last few messages (which expected no output) were just not being processed at all.

### CODE REVIEW

**Review by christopherfujino — DISMISSED** [review]
You have some analysis errors, but otherwise LGTM

**Comment by Hixie** [comment]
@christopherfujino PTAL, i added some changes to the CI stuff to make the error messages i was getting more useful

File: dev/bots/run_command.dart [hunk 1]
```diff
@@ -190,13 +190,24 @@ Future<CommandResult> runCommand(String executable, List<String> arguments, {
         print(result.flattenedStderr);
         break;
     }
+    String allOutput;
+    if (failureMessage == null) {
+      allOutput = '${result.flattenedStdout}\n${result.flattenedStderr}';
+      if (allOutput.split('\n').length > 10) {
```
Comment (christopherfujino) [inline dev/bots/run_command.dart:196]: What about (couldn't figure out how to use the GitHub suggestion API with a multi-line diff):

```
      final List<String> allOutputLines = allOutput.split('\n');
      if (allOutputLines.length > 10) {
        allOutput = <String>[
          ...allOutputLines.take(10),
          '(more stdout/stderr lines truncated)',
        ].join('\n');
      }
```

Comment (christopherfujino) [reply dev/bots/run_command.dart:196]: This is a nit suggestion, approving regardless whether you take it

**Review by christopherfujino — APPROVED** [review]

File: dev/bots/run_command.dart (hunk 1 above)
Comment (Hixie) [reply dev/bots/run_command.dart:196]: I figured if there's that much output, I have no way to guess if the useful part is at the top or bottom (or neither) so...

Comment (christopherfujino) [reply dev/bots/run_command.dart:196]: You're right, I figured either way you have a 50% chance you got something interesting :)

**Comment by Hixie** [comment]
FWIW i made one (hopefully) last tweak:

```diff
   print('${red}For your convenience, the error messages reported above are repeated here:$reset');
-  print('  🙙  🙛  ');
+  final bool printSeparators = _errorMessages.any((List<String> messages) => messages.length > 1);
+  if (printSeparators) {
+    print('  🙙  🙛  ');
+  }
   for (int index = 0; index < _errorMessages.length * 2 - 1; index += 1) {
     if (index.isEven) {
       _errorMessages[index ~/ 2].forEach(print);
-    } else {
+    } else if (printSeparators) {
       print('  🙙  🙛  ');
     }
   }
```

I did this because I noticed that sometimes the output you get here is just lots of one-line errors and it would be a mess if they all got separated.

**Pull request merged**